plugins/org.eclipse.osee.framework.skynet.core/src/org/eclipse/osee/framework/skynet/core/importing/parsers/MsoApplicationExtractor.java

```diff
@@ -0,0 +1,126 @@
+/*********************************************************************
+ * Copyright (c) 2024 Boeing
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Boeing - initial API and implementation
+ **********************************************************************/
+package org.eclipse.osee.framework.skynet.core.importing.parsers;
+
+import java.io.BufferedReader;
+import org.eclipse.osee.framework.core.data.MicrosoftOfficeApplicationEnum;
+
+/**
+ * Extracts the value of the "mso-application" processing instruction from an XML document. This helps identify the
+ * associated Microsoft Office application (e.g., Word, Excel).
+ * <p>
+ * The main method, {@link #findMsoApplicationValue(BufferedReader)}, retrieves the "progid" value of the
+ * "mso-application" processing instruction if present.
+ * </p>
+ * <p>
+ * Note: This implementation scans the XML content for the specific processing instruction rather than parsing the
+ * entire XML structure.
+ * </p>
+ *
+ * @author Jaden W. Puckett
+ */
+public class MsoApplicationExtractor {
+
+   /**
+    * Extracts the "mso-application" progid value from the provided XML content.
+    * <p>
+    * This method scans the XML content for a processing instruction like {@code <?mso-application progid="...">}. If
+    * found, the "progid" value is returned as a {@link MicrosoftOfficeApplicationEnum}. If no such instruction exists,
+    * an {@code IllegalArgumentException} is thrown.
+    * </p>
+    *
+    * @param xmlContent a {@link BufferedReader} containing the XML content to process; must not be {@code null}.
```

Comment:
what are all the @link for?  the method signature has BufferedReader that the user can go visit if they're interested, not sure why you need another link in javadoc.